TTVA-228 | Upgrade Django, Python & Node

japauliina opened the pull request:
Upgrade major tech stack versions:
- Upgrade Django to 4.2
- Upgrade Python to 3.10
- Upgrade Node.js to v20

Refs TTVA-228, TTVA-232, TTVA-233, TTVA-234, TTVA-235

japauliina committed:
Change JSONField to models.JSONField

Change django.contrib.postgres.fields.JSONField to models.JSONField  as
it will be removed in Django 4.0.

Refs TTVA-232

japauliina committed:
Upgrade Django to 3.0

Upgrade Django to 3.0 and update also other dependencies to the latest
possible versions.

Changes:
- Replace deprecated ugettext_lazy with gettext_lazy
- Replace django.utils.six with six

Refs TTVA-232

japauliina committed:
Fix RemovedInDjango40Warnings

Replace deprecated functions with the new ones:
- force_text -> force_str
- ugettext -> gettext
- ungettext -> ngettext

Refs TTVA-232

japauliina committed:
Upgrade Django to 3.1

Upgrade Django to 3.1 and update also other dependencies to the latest
possible versions.

Changes:
- New migration for AbstractUser.first_name max_length increased to 150
- Fix FieldDoesNotExist import as it was removed from
django.db.models.fields

Refs TTVA-232

japauliina committed:
Upgrade Django to 3.2

Upgrade Django to 3.2 and update also other dependencies to the latest
possible versions.

Changes:
- Get response headers from HttpResponse.headers as they are now stored
there.
- Add DEFAULT_AUTO_FIELD setting as suggested by Django 3.2.
- Remove usage of QuerySet.distinct() after union() as it is not
supported anymore.

Refs TTVA-232

japauliina committed:
Change NullBooleanField to BooleanField

NullBooleanField is deprecated and will be removed in Django 4.0.
Change to suggested BooleanField(null=True).

Refs TTVA-232

japauliina committed:
Upgrade Django to 4.0

Upgrade Django to 4.0 and update also other dependencies to the latest
possible versions.

USE_DEPRECATED_PYTZ is set to True for now. Django is migrating from
pytz to zoneinfo and this setting will be removed in Django 5.0.

Changes:
- Migrations autodetector changes
- Remove providing_args argument from django.dispatch.Signal
- Replace removeddjango.conf.urls.url with django.urls.re_path
- Add now required length argument to get_random_string

Refs TTVA-232

japauliina committed:
Upgrade Django to 4.1

Upgrade Django to 4.1 and update also other dependencies to the latest
possible versions.

Refs TTVA-232

japauliina committed:
Upgrade Django to 4.2

Upgrade Django to 4.2 and update also other dependencies to the latest
possible versions.

Changes:
- TranslationOptions's fields was named to all_fields
- Fix error "Router with basename "resource" is already registered" by
adding ResourceListViewSet directly to the urlpatterns.

Refs TTVA-232

japauliina committed:
Upgrade Python to 3.9

Refs TTVA-233

japauliina committed:
Upgrade Node.js to v16

Refs TTVA-234

japauliina committed:
Upgrade Node.js to v18

Refs TTVA-234

japauliina committed:
Upgrade Node.js to v20

Refs TTVA-234

japauliina committed:
Remove unused hel-icons package

Refs TTVA-234

japauliina committed:
Remove unused js-yaml package

Refs TTVA-234

japauliina committed:
Remove unused @popperjs/core package

Refs TTVA-234

japauliina committed:
Remove unused uuid package

Refs TTVA-234

japauliina committed:
Remove commented out code

Refs TTVA-234

japauliina committed:
Update packages

Refs TTVA-234

japauliina committed:
Remove django-ckeditor

Django-ckeditor is bundled with CKEditor 4.22.1 which isn't supported
anymore and has unfixed security issues. Remove django-ckeditor and
switch to a TextField for now.

Refs TTVA-235

japauliina committed:
Upgrade Python to 3.10

Refs TTVA-233

japauliina committed:
Remove obsolete version attribute from docker-compose.yml

jopesy reviewed (approved):
Looks very good, and this was nice to review as well thanks to the nicely split commits and commit messages. Also I approve of the fact the number of added lines is an even 10000 which looks nice. Good work Pauliina!

![image](https://github.com/user-attachments/assets/6bf5f6fa-5375-4c2d-a08e-726a02d39d65)

jorilindell committed:
Upgrade Python to 3.12

jorilindell committed:
Upgrade Node.js to v24

jorilindell committed:
Fix font asset handling in webpack configuration

Replace deprecated file-loader with webpack 5's built-in asset/resource
for font files (.woff, .woff2, .ttf, .eot, .svg). This resolves an issue
where font files were being treated as JavaScript modules that export
paths instead of actual binary font assets.

The previous file-loader configuration caused browsers to receive
JavaScript code when requesting font files, breaking icon font rendering
(Glyphicons). The new asset/resource approach properly
emits fonts as static assets while maintaining the same output directory
structure.

This change is necessary for webpack 5 compatibility and ensures proper
font loading in the browser.
